### PR TITLE
Add option to download Fragment Analyser/Caliper images as zip files

### PR DESCRIPTION
**Pull request by aanil**
WIP: Not sure of the zip download method in js

### CODE REVIEW

**Review by alneberg — CHANGES_REQUESTED** [review]
Sadly didn't get this to work. The zip archive was empty when it finally was downloaded even though it was not supposed to be. Really like the design of it though with the placement of the button and the look of the modal.

**Comment by alneberg** [comment]
Ah, sorry the project didn't have that kind of images. I guess we should have some kind of check for that as well. Perhaps even tell the user how many images are available of each kind before the download? Would it be too complicated?

**Review by alneberg — APPROVED** [review]
Ok, I managed to get it working now, only small possible improvements: The "downloading..." button never changes back after the download has finished. I guess that could be an issue if one wants to download more than one type of images? Not sure if that happens though. The second improvement would be if one could know how many images are available of each kind before downloading? But that's also not necessary.

File: status/projects.py [hunk 1]
```diff
@@ -588,6 +588,62 @@ def get_caliper_image(url):
...
+        fileName = '{}_{}_images.zip'.format(project, name)
+        f = BytesIO()
+        with zp.ZipFile(f, "w") as zf:
+            for sample in data:
+                if data[sample]:
+                    for key in data[sample]:
+                        img_file_name = sample + '.'
+                        if 'frag_an' in type and 'initial_qc' in key:
+                            img_file_name = img_file_name + 'png'
+                            print(img_file_name)
```
[the remark this answers is not in the document]
Comment (aanil) [reply status/projects.py]: Removed it!

File: run_dir/static/js/running_notes.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ function generate_category_label(category){
          category='<span class="label label-warning">'+ category +"</span>";
      }else if (category == 'Important'){
          category='<span class="label label-imp">'+ category +"</span>";
-     }else if (category == 'Deviation'){	
+     }else if (category == 'Deviation'){
```
Comment (alneberg) [inline run_dir/static/js/running_notes.js:23]: I choose not to see that you've added this file change

Comment (aanil) [reply run_dir/static/js/running_notes.js:23]: Atom added it to the file and I added it to the PR by mistake 😅

File: status/projects.py [hunk 3]
```diff
@@ -588,6 +588,64 @@ def get_caliper_image(url):
...
+        try:
+            data = result.rows[0].value
+        except TypeError:
+            #can be triggered by the data.get().get() calls.
+            raise tornado.web.HTTPError(404, reason='No caliper image found')
+
+        fileName = '{}_{}_images.zip'.format(project, name)
+        f = BytesIO()
+        num_files = 0
+        with zp.ZipFile(f, "w") as zf:
```
Comment (alneberg) [inline status/projects.py:624]: I think you could do `with zp.ZipFile(f, "w") as zf, BytesIO() as f:` should save you a bit of precious lines of code, but it's also more certain that the close is always called right?

**Review by alneberg — APPROVED** [review]
Working fine now! I guess the error handling can be refined, but it would be a bit more work, so let's get back to that later if necessary.

File: status/projects.py (hunk 3 above)
Comment (aanil) [reply status/projects.py:624]: I had some issues with corrupted zip files when I tried it. We can put it on the list of improvements?

**Pull request merged**